velox/functions/sparksql/tests/AtLeastNNonNullsTest.cpp

```diff
@@ -0,0 +1,142 @@
+  }
+
+  void testAtLeastNNonNulls(
+      int32_t n,
+      const std::vector<VectorPtr>& input,
+      const VectorPtr& expected) {
+    const auto result = atLeastNNonNulls<int32_t>(n, input);
+    assertEqualVectors(expected, result);
+  }
+};
+
+TEST_F(AtLeastNNonNullsTest, basic) {
+  auto stringInput = makeNullableFlatVector<StringView>(
+      {std::nullopt, "1", "", std::nullopt, ""});
+  auto boolInput = makeNullableFlatVector<bool>(
+      {std::nullopt, true, false, std::nullopt, std::nullopt});
+  auto intInput =
+      makeNullableFlatVector<int32_t>({-1, 0, 1, std::nullopt, std::nullopt});
+  auto floatInput = makeNullableFlatVector<float>(
+      {FloatConstants::kMaxFloat,
+       FloatConstants::kNaNFloat,
+       0.1f,
+       0.0f,
+       std::nullopt});
+  auto doubleInput = makeNullableFlatVector<double>(
+      {std::log(-2.0),
+       FloatConstants::kMaxDouble,
+       FloatConstants::kNaNDouble,
+       std::nullopt,
+       0.1});
+  auto arrayInput = makeArrayVectorFromJson<int32_t>(
+      {"[1, null, 3]", "[1, 2, 3]", "null", "[null]", "[]"});
+  auto mapInput = makeMapVectorFromJson<int32_t, int32_t>(
+      {"{1: 10, 2: null, 3: null}", "{1: 10, 2: 20}", "{1: 2}", "{}", "null"});
+  auto constInput = makeConstant<int32_t>(2, 5);
+  auto indices = makeIndices(5, [](auto row) { return (row + 1) % 5; });
+  auto dictInput = wrapInDictionary(indices, 5, doubleInput);
+
+  auto expected = makeFlatVector<bool>({false, true, true, false, false});
+  testAtLeastNNonNulls(2, {stringInput, boolInput}, expected);
+
+  expected = makeFlatVector<bool>({false, false, false, false, false});
+  testAtLeastNNonNulls(3, {stringInput, boolInput}, expected);
+
+  expected = makeFlatVector<bool>({true, true, true, true, true});
+  testAtLeastNNonNulls(0, {stringInput, boolInput}, expected);
+  testAtLeastNNonNulls(-1, {stringInput, boolInput}, expected);
+
+  expected = makeFlatVector<bool>({true, false, true, true, false});
+  testAtLeastNNonNulls(1, {floatInput}, expected);
+
+  expected = makeFlatVector<bool>({false, true, false, false, true});
+  testAtLeastNNonNulls(1, {doubleInput}, expected);
+
+  expected = makeFlatVector<bool>({false, true, true, false, false});
+  testAtLeastNNonNulls(2, {stringInput, boolInput, floatInput}, expected);
+
+  expected = makeFlatVector<bool>({false, true, true, false, false});
+  testAtLeastNNonNulls(
+      3, {boolInput, intInput, floatInput, doubleInput}, expected);
+
+  expected = makeFlatVector<bool>({false, false, false, false, false});
+  testAtLeastNNonNulls(2, {floatInput, doubleInput}, expected);
+
+  expected = makeFlatVector<bool>({true, false, false, true, false});
+  testAtLeastNNonNulls(
+      4, {mapInput, arrayInput, constInput, dictInput}, expected);
+}
+
+TEST_F(AtLeastNNonNullsTest, error) {
+  auto input =
+      makeNullableFlatVector<int32_t>({-1, 0, 1, std::nullopt, std::nullopt});
+
+  VELOX_ASSERT_USER_THROW(
+      atLeastNNonNulls<double>(1.0, {input}),
```

Comment:
These tests would be easier to read if written as 

```
VELOX_ASSERT_USER_THROW(
   evaluate("at_least_n_non_null(1.0, c0)", input),
   ...
```

Then, we won't need template parameter for atLeastNNonNulls helper method.

Reply:
Done.